src/code/audio/code_8019AF00.c

```diff
@@ -4463,35 +4743,238 @@ void Audio_ClearSariaBgm2(void) {
+    }
+}
+
+s32 Audio_IsSequencePlaying(u8 seqId) {
+    u8 playerIndex = SEQ_PLAYER_BGM_MAIN;
+
+    if (sSeqFlags[seqId & 0xFF] & SEQ_FLAG_FANFARE) {
+        playerIndex = SEQ_PLAYER_FANFARE;
+    } else if (sSeqFlags[seqId & 0xFF] & SEQ_FLAG_FANFARE_KAMARO) {
+        playerIndex = SEQ_PLAYER_FANFARE;
+    }
+
+    if (seqId == (Audio_GetActiveSequence(playerIndex) & 0xFF)) {
+        return true;
+    } else {
+        return false;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/code_8019AF00/func_801A2D54.s")
+void Audio_PlayBgm_StorePrevBgm(u16 seqId) {
+    u16 curSeqId = Audio_GetActiveSequence(SEQ_PLAYER_BGM_MAIN);
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/code_8019AF00/Audio_IsSequencePlaying.s")
+    if (curSeqId == NA_BGM_DISABLED) {
+        curSeqId = NA_BGM_GENERAL_SFX;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/code_8019AF00/func_801A2E54.s")
+    if (curSeqId != seqId) {
+        Audio_SetSequenceMode(SEQ_MODE_IGNORE);
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/code_8019AF00/func_801A2ED8.s")
+        // Ensure the sequence about to be stored isn't also storing a separate sequence
+        if (!(sSeqFlags[curSeqId] & SEQ_FLAG_RESTORE)) {
+            sPrevMainBgmSeqId = curSeqId;
+        }
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/code_8019AF00/func_801A2F88.s")
+        Audio_QueueSeqCmd(0x00000000 | ((u32)(SEQ_PLAYER_BGM_MAIN) << 24) | ((u32)(0) << 0x10) | (u32)(seqId + 0x8000));
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/code_8019AF00/func_801A2FC4.s")
+/**
+ * To be used in conjunction with Audio_PlayBgm_StorePrevBgm
+ */
+void Audio_RestorePrevBgm(void) {
+    if ((Audio_GetActiveSequence(SEQ_PLAYER_BGM_MAIN) != NA_BGM_DISABLED) &&
+        (sSeqFlags[Audio_GetActiveSequence(SEQ_PLAYER_BGM_MAIN) & 0xFF] & SEQ_FLAG_RESTORE)) {
+        if ((sPrevMainBgmSeqId == NA_BGM_DISABLED) || (sPrevMainBgmSeqId == NA_BGM_GENERAL_SFX)) {
+            Audio_QueueSeqCmd(0x100000FF | ((u8)(SEQ_PLAYER_BGM_MAIN) << 24) | ((0) << 16));
+        } else {
+            if (sPrevMainBgmSeqId == NA_BGM_AMBIENCE) {
+                sPrevMainBgmSeqId = sPrevAmbienceSeqId;
+            }
+            Audio_QueueSeqCmd(0x00000000 | ((u32)(SEQ_PLAYER_BGM_MAIN) << 24) | ((u32)(0) << 0x10) |
+                              (u32)(sPrevMainBgmSeqId + 0x8000));
+        }
+        sPrevMainBgmSeqId = NA_BGM_DISABLED;
+    }
+}
+
+// Unused
+void Audio_PlayAmbience_StorePrevBgm(u8 ambienceId) {
+    u16 seqId = Audio_GetActiveSequence(SEQ_PLAYER_BGM_MAIN);
+
+    if (seqId != NA_BGM_AMBIENCE) {
+        sPrevMainBgmSeqId = seqId;
+    }
+
+    Audio_PlayAmbience(ambienceId);
+}
+
+// Unused
+void Audio_ForceRestorePreviousBgm(void) {
+    if (sPrevMainBgmSeqId != NA_BGM_DISABLED) {
+        Audio_QueueSeqCmd(0x00000000 | ((u32)(SEQ_PLAYER_BGM_MAIN) << 24) | ((u32)(0) << 0x10) |
+                          (u32)(sPrevMainBgmSeqId + 0x8000));
+    }
+
+    sPrevMainBgmSeqId = NA_BGM_DISABLED;
+}
```

Comment:
These are leftover from OoT and appear to be designed specifically for the nature ambience, since in OoT shares seqPlayerIndex 0 with both the main bgm and the ambience. So anytime the ambience track comes on in OoT, the main bgm would need to temporarily cut out. But if you wanted to restore the main bgm track (think playing Song of Storms), then these are the functions you would use.

In MM, they introduced a whole new seqPlayerIndex exclusively for the ambience track (seqPlayerIndex = 5). That means MM can play both ambience and the main bgm at the same time, meaning these functions were no longer needed.